tensorflow/core/ops/training_ops.cc

```diff
@@ -102,6 +102,30 @@ use_locking: If `True`, the subtraction will be protected by a lock;
   otherwise the behavior is undefined, but may exhibit less contention.
 )doc");
 
+REGISTER_OP("ApplyDelayCompensatedGradientDescent")
```

Comment:
Sorry for not seeing this before: can you update this to use the "resource type" like in ResourceApplyGradientDescent instead?  We're moving to "Resource variables" which have meaningful type semantics (refs don't).